src/libraries/Microsoft.PowerFx.Interpreter/Environment/PowerFxConfig.cs

```diff
@@ -0,0 +1,80 @@
+﻿// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+using System;
+using System.Collections.Generic;
+using System.Globalization;
+using Microsoft.PowerFx.Core;
+using Microsoft.PowerFx.Core.Functions;
+using Microsoft.PowerFx.Core.Types.Enums;
+
+namespace Microsoft.PowerFx
+{
+    /// <summary>
+    /// A container that allows for compiler customization.
+    /// </summary>
+    public sealed class PowerFxConfig
+    {
+        private bool _isLocked;
+        private readonly Dictionary<string, TexlFunction> _extraFunctions;
+
+        internal ImmutableEnvironmentSymbolTable ImmutableEnvironmentSymbolTable;
```

Comment:
>ImmutableEnvironmentSymbolTable ImmutableEnvironmentSymbolTable; [](http://example.com/codeflow?start=17&length=64)

The Immutable here is out of place next to all the mutable things. 

Could this just be: 
Dictionary<DName, IExternalEntity> ?